src/backend/core/cms/views.py

```diff
@@ -36,7 +36,10 @@ def get(self, request):
         """
         Gets all pages.
         """
-        pages = Page.objects.all()
+
+        pages = Page.objects.all().exclude(
+            safe_deleted=True
+        )  # even though we have a custom manager, we still need to exclude soft deleted objects here because of polymorphic
```

Comment:
If you swap the order of base classes for `Page` (put `SafeDeleteModel` first), does it work?

Reply:
Sadly doesn't help 😢 I think I tried most of different combinations incl. custom managers, making `SafeDeleteModel` the lower level models (`PageCMS`, ...)  but none of that helps.